<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>764</width>
    <height>652</height>
   </rect>
  </property>
  <property name="acceptDrops">
   <bool>true</bool>
  </property>
  <property name="windowTitle">
   <string>Super Posterize</string>
  </property>
  <property name="unifiedTitleAndToolBarOnMac">
   <bool>true</bool>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <property name="spacing">
     <number>0</number>
    </property>
    <item row="0" column="1">
     <widget class="QGraphicsView" name="preview">
      <property name="acceptDrops">
       <bool>false</bool>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QWidget" name="widget" native="true">
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QGroupBox" name="settings_LimitInputSize">
         <property name="title">
          <string>Limit Input Size</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <layout class="QFormLayout" name="formLayout_5">
          <item row="0" column="0">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Max Input Size</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="input_MaxInputSize">
            <property name="suffix">
             <string> px</string>
            </property>
            <property name="minimum">
             <number>64</number>
            </property>
            <property name="maximum">
             <number>2048</number>
            </property>
            <property name="singleStep">
             <number>64</number>
            </property>
            <property name="value">
             <number>256</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="settings_Downscale">
         <property name="title">
          <string>&amp;Downscale</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
         <layout class="QFormLayout" name="formLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Factor</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="input_DownscaleFactor">
            <property name="suffix">
             <string/>
            </property>
            <property name="prefix">
             <string>1/</string>
            </property>
            <property name="minimum">
             <number>2</number>
            </property>
            <property name="maximum">
             <number>16</number>
            </property>
            <property name="value">
             <number>2</number>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Method</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QComboBox" name="input_ScalingMethod">
            <item>
             <property name="text">
              <string>AVIR</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>DPID</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Bilinear</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>DPID Sharpening Curve</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QDoubleSpinBox" name="input_SharpeningCurve">
            <property name="prefix">
             <string>to the power of </string>
            </property>
            <property name="maximum">
             <double>5.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.100000000000000</double>
            </property>
            <property name="value">
             <double>0.500000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="settings_AlphaThreshold">
         <property name="title">
          <string>Alpha Threshold</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
         <layout class="QFormLayout" name="formLayout_4">
          <item row="0" column="1">
           <widget class="QSlider" name="input_AlphaThreshold">
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="value">
             <number>50</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="tickPosition">
             <enum>QSlider::TicksAbove</enum>
            </property>
            <property name="tickInterval">
             <number>25</number>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>Threshold</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="settings_NormalizeLuminance">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="title">
          <string>&amp;Normalized Grayscale</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
         <layout class="QFormLayout" name="formLayout_2">
          <item row="0" column="0">
           <widget class="QLabel" name="label_MinPoint">
            <property name="text">
             <string>Black</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QDoubleSpinBox" name="input_BlackPoint">
            <property name="suffix">
             <string/>
            </property>
            <property name="decimals">
             <number>3</number>
            </property>
            <property name="maximum">
             <double>0.200000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.010000000000000</double>
            </property>
            <property name="value">
             <double>0.050000000000000</double>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_MidPoint">
            <property name="text">
             <string>Gray</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QDoubleSpinBox" name="input_GrayPoint">
            <property name="decimals">
             <number>3</number>
            </property>
            <property name="minimum">
             <double>0.200000000000000</double>
            </property>
            <property name="maximum">
             <double>0.900000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.050000000000000</double>
            </property>
            <property name="value">
             <double>0.650000000000000</double>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_MaxPoint">
            <property name="text">
             <string>White</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QDoubleSpinBox" name="input_WhitePoint">
            <property name="decimals">
             <number>3</number>
            </property>
            <property name="minimum">
             <double>0.900000000000000</double>
            </property>
            <property name="maximum">
             <double>0.999000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.010000000000000</double>
            </property>
            <property name="value">
             <double>0.990000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="settings_Posterize">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="title">
          <string>Posteri&amp;ze</string>
         </property>
         <property name="flat">
          <bool>false</bool>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
         <layout class="QFormLayout" name="formLayout_3">
          <item row="0" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Hue</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="input_MaterialSteps">
            <property name="suffix">
             <string> steps</string>
            </property>
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="value">
             <number>8</number>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Luminosity</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="input_LuminanceSteps">
            <property name="suffix">
             <string> steps</string>
            </property>
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>128</number>
            </property>
            <property name="value">
             <number>8</number>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLabel" name="label_TotalColors">
            <property name="text">
             <string>TextLabel</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Total Colors</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>764</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>Fi&amp;le</string>
    </property>
    <addaction name="actionLoad"/>
    <addaction name="actionSave"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <action name="actionLoad">
   <property name="text">
    <string>&amp;Load</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>&amp;Save</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>settings_Downscale</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>settingsChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>131</x>
     <y>72</y>
    </hint>
    <hint type="destinationlabel">
     <x>262</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>settings_NormalizeLuminance</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>settingsChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>131</x>
     <y>156</y>
    </hint>
    <hint type="destinationlabel">
     <x>262</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>settings_Posterize</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>settingsChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>131</x>
     <y>321</y>
    </hint>
    <hint type="destinationlabel">
     <x>262</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>input_MaterialSteps</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>settingsChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>162</x>
     <y>303</y>
    </hint>
    <hint type="destinationlabel">
     <x>262</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>input_DownscaleFactor</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>settingsChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>149</x>
     <y>69</y>
    </hint>
    <hint type="destinationlabel">
     <x>262</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>input_LuminanceSteps</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>settingsChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>162</x>
     <y>332</y>
    </hint>
    <hint type="destinationlabel">
     <x>262</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveImageAction()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>262</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLoad</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>loadImageAction()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>262</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>input_MaxInputSize</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>settingsChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>169</x>
     <y>68</y>
    </hint>
    <hint type="destinationlabel">
     <x>262</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>input_AlphaThreshold</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>settingsChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>157</x>
     <y>186</y>
    </hint>
    <hint type="destinationlabel">
     <x>262</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>settings_AlphaThreshold</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>settingsChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>131</x>
     <y>179</y>
    </hint>
    <hint type="destinationlabel">
     <x>262</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>input_BlackPoint</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>settingsChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>158</x>
     <y>227</y>
    </hint>
    <hint type="destinationlabel">
     <x>262</x>
     <y>217</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>input_GrayPoint</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>settingsChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>158</x>
     <y>245</y>
    </hint>
    <hint type="destinationlabel">
     <x>262</x>
     <y>217</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>input_WhitePoint</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>settingsChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>158</x>
     <y>263</y>
    </hint>
    <hint type="destinationlabel">
     <x>262</x>
     <y>217</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>input_ScalingMethod</sender>
   <signal>activated(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>settingsChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>169</x>
     <y>110</y>
    </hint>
    <hint type="destinationlabel">
     <x>262</x>
     <y>217</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>input_SharpeningCurve</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>settingsChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>250</x>
     <y>140</y>
    </hint>
    <hint type="destinationlabel">
     <x>381</x>
     <y>325</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>settingsChanged()</slot>
  <slot>saveImageAction()</slot>
  <slot>loadImageAction()</slot>
 </slots>
</ui>
